azurerm/resource_arm_app_service_test.go

```diff
@@ -1182,6 +1182,36 @@ func TestAccAzureRMAppService_minTls(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMAppService_corsSettings(t *testing.T) {
+	resourceName := "azurerm_app_service.test"
+	ri := tf.AccRandTimeInt()
+	config := testAccAzureRMAppService_corsSettings(ri, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAppServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAppServiceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.0.support_credentials", "true"),
+					resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.0.allowed_origins.#", "3"),
+					resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.0.allowed_origins.0", "http://www.contoso.com"),
+					resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.0.allowed_origins.1", "www.contoso.com"),
+					resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.0.allowed_origins.2", "contoso.com"),
```

Comment:
changing this to a Set will mean these values change based on their hash value, as such we'll need to update these:

```
resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.0.allowed_origins.376798695", "http://www.contoso.com"),
resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.0.allowed_origins.1456776878", "www.contoso.com"),
resource.TestCheckResourceAttr(resourceName, "site_config.0.cors.0.allowed_origins.29876578", "contoso.com"),
```

(whilst you could probably work these out by hand, we generally spin this up manually / run it once to confirm what the values should be, since the hash function is tested elsewhere)